### PR TITLE
Fixes Issue 5006

### LINKED ISSUE #5006: MultiBinding NRE
null reference exception when using MultiBinding without StringFormat and Converter

```xaml
        <StackPanel>
          <TextBlock Text="{Binding Position,Mode=OneWay}"/>
          <Grid ColumnDefinitions="Auto,*" RowDefinitions="Auto,Auto,Auto">
            <TextBlock Text="X"/>
            <NumericUpDown x:Name="x"
                           Grid.Column="1"/>
            <TextBlock Text="Y"
                       Grid.Row="1"/>
            <NumericUpDown x:Name="y"
                       Grid.Row="1"
                       Grid.Column="1"/>
          </Grid>
          <Button Content="Set Position" Command="{Binding SetPosition}">
            <Button.CommandParameter>
              <MultiBinding>
                <Binding ElementName="x" Path="Value"/>
                <Binding ElementName="y" Path="Value"/>
              </MultiBinding>
            </Button.CommandParameter>
          </Button>
        </StackPanel>
```

### PR DESCRIPTION
**Pull request by workgroupengineering**
## What does the pull request do?

Avoid null reference exception when using MultiBinding without StringFormat and Converter and retrun array of object

## What is the current behavior?

throe NRE

## Checklist

- [x] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

## Fixed issues

Fixes #5006

### CODE REVIEW

File: src/Markup/Avalonia.Markup/Data/MultiBinding.cs [hunk 1]
```diff
@@ -116,7 +116,13 @@ private object ConvertValue(IList<object> values, Type targetType, IMultiValueCo
             }
 
             var culture = CultureInfo.CurrentCulture;
-            var converted = converter.Convert(values, targetType, ConverterParameter, culture);
+            object converted;
+            if (converter != null)
+            {
+                converted = converter.Convert(values, targetType, ConverterParameter, culture);
+            }
+            else
```
Comment (MarchingCube) [inline src/Markup/Avalonia.Markup/Data/MultiBinding.cs:124]: Braces please :)

File: src/Markup/Avalonia.Markup/Data/MultiBinding.cs [hunk 2]
```diff
@@ -116,7 +116,13 @@ private object ConvertValue(IList<object> values, Type targetType, IMultiValueCo
...
+                converted = values.ToArray();
```
Comment (MarchingCube) [inline src/Markup/Avalonia.Markup/Data/MultiBinding.cs:125]: Is there a reason why you are copying this list?

File: tests/Avalonia.Markup.UnitTests/Extensions/IEnummerableExtension.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+using System.Collections;
+
+namespace Avalonia.Markup.UnitTests
+{
+    static class IEnummerableExtension
+    {
+        public static object ElementAt(this IEnumerable source, int index)
+        {
+            var i = -1;
+            var enumerator = source.GetEnumerator();
```
Comment (MarchingCube) [inline tests/Avalonia.Markup.UnitTests/Extensions/IEnummerableExtension.cs:11]:
```suggestion
            using var enumerator = source.GetEnumerator();
```

Comment (workgroupengineering) [reply tests/Avalonia.Markup.UnitTests/Extensions/IEnummerableExtension.cs:11]: IEnumerator can not implicitly cast to System.IDisposable

File: src/Markup/Avalonia.Markup/Data/MultiBinding.cs (hunk 2 above)
Comment (workgroupengineering) [reply src/Markup/Avalonia.Markup/Data/MultiBinding.cs:125]: I don't want to allow values.add to be used for consistency reasons

File: tests/Avalonia.Markup.UnitTests/Extensions/IEnummerableExtension.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections;
+
+namespace Avalonia.Markup.UnitTests
+{
+    static class IEnummerableExtension
```
Comment (MarchingCube) [inline tests/Avalonia.Markup.UnitTests/Extensions/IEnummerableExtension.cs:6]:
```suggestion
   internal static class IEnumerableExtensions
```

Comment (MarchingCube) [reply tests/Avalonia.Markup.UnitTests/Extensions/IEnummerableExtension.cs:6]: I didn’t notice this previously.

**Review by MarchingCube — APPROVED** [review]
Looks fine to me now, thanks :)

**Pull request merged**